transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java

```diff
@@ -383,6 +389,23 @@ private static InputStream maybeUnGzip(InputStream input, String contentEncoding
     return input;
   }
 
+  private String findPseudoId(BackendRequest request) {
+    Iterator<EventInternal> events = request.getEvents().iterator();
+
+    if(!events.hasNext()) return null;
+
+    String pseudoId = events.next().getPseudonymousId();
+
+    for(EventInternal event : request.getEvents()) {
+      String currentId = event.getPseudonymousId();
+      if(!Objects.equals(pseudoId, currentId)) {
```

Comment:
this is identity instead of equality, intended?

Reply:
I thought so too, but it's actually not:
```java
public static boolean equals(Object a, Object b) {
  return (a == b) || (a != null && a.equals(b));
}
```